src/compile.js

```diff
@@ -430,12 +430,14 @@ Compiler.prototype.cdict = function (e) {
     var v;
     var i;
     var items;
-    Sk.asserts.assert(e.values.length === e.keys.length);
     items = [];
-    for (i = 0; i < e.values.length; ++i) {
-        v = this.vexpr(e.values[i]); // "backwards" to match order in cpy
-        items.push(this.vexpr(e.keys[i]));
-        items.push(v);
+    if (e.keys !== null) {
```

Comment:
How do you feel about this @meredydd we could also catch these cases in `visitExpr` and be more specific about when we accept `null` as valid.

Reply:
So, there are a few changes here that cope with `null` inputs; being slightly afraid of the parser I have no idea when it will generate `null`s. My default would therefore be to accept whatever the C version accepts, and `assert` wherever we're not

Reply:
So I think it's correct that the parser now outputs `null` instead of an empty array. I'll try and find the corresponding c code and see what how they handle it.